### PR TITLE
[MAINT] remove dependency on matplotlib to filter packages to be tested

### LINKED ISSUE #4702: remove dependency on matplotlib for selection of tests to run
> Is there a condition somewhere that excludes this package when `matplotlib` is not installed?

I think this part of the conftest.py is what does this.

https://github.com/nilearn/nilearn/blob/6a7a8078ef4a26e2d49203eb599cb0e395808cfe/nilearn/conftest.py#L27-L38

_Originally posted by @Remi-Gau in https://github.com/nilearn/nilearn/pull/4675#discussion_r1814745799_

### PR DESCRIPTION
**Pull request by hndgzkn**
- Closes #4702.

- tests in `nilearn.plotting` and `nilearn.reporting` are still ignored by `conftest.py` 
- removed all `@pytest.mark.skipif(not is_matplotlib_installed(), ...` lines  from `nilearn.plotting` and `nilearn.reporting` tests, as these tests are already ignored by `conftest.py` (considering [your answer](https://github.com/nilearn/nilearn/issues/4702#issuecomment-2456859036) on reporting @Remi-Gau  to be able to work without matplotlib in the future, I am not sure about this part, and I can revert it. However for the moment I see these marks misleading)
- moved `_set_mpl_backend` to `nilearn._utils.helpers` so that it can be imported from any package
- added tests to see if the exception and warning are raised when matplotlib is not installed while importing from `nilearn.plotting` and `nilearn.reporting`

### CODE REVIEW

File: nilearn/_utils/testing.py [hunk 1]
```diff
@@ -172,7 +172,8 @@ def write_imgs_to_path(*imgs, file_path=None, **kwargs):
 
 def are_tests_running():
     """Return whether we are running the pytest test loader."""
-    return "PYTEST_CURRENT_TEST" in os.environ
```
Comment (hndgzkn) [inline nilearn/_utils/testing.py:175]: This check did not work.

File: nilearn/plotting/__init__.py [hunk 2]
```diff
@@ -1,61 +1,7 @@
...
-def _set_mpl_backend():
-    # We are doing local imports here to avoid polluting our namespace
-    try:
-        import matplotlib
-    except ImportError:
-        if importlib.util.find_spec("pytest") is not None:
-            from .._utils.testing import skip_if_running_tests
-
-            # No need to fail when running tests
-            skip_if_running_tests("matplotlib not installed")
```
Comment (hndgzkn) [inline nilearn/plotting/__init__.py:23]: I've removed these lines while moving the function to `nilearn._utils.helpers.py` as the tests of plotting package is already ignored by `conftest.py`

**Comment by Remi-Gau** [comment]
I had very quick first look and things look good so far.
Will do another more precise pass when CI finishes.

File: nilearn/_utils/helpers.py [hunk 3]
```diff
@@ -3,6 +3,47 @@
 import os
 import warnings
 
+OPTIONAL_MATPLOTLIB_MIN_VERSION = "3.3.0"
+
+
+def _set_mpl_backend():
```
Comment (bthirion) [inline nilearn/_utils/helpers.py:9]: Please include a one-line docstring

**Review by bthirion — COMMENTED** [review]
LGTM overall, thx.

**Comment by Remi-Gau** [comment]
> * removed all `@pytest.mark.skipif(not is_matplotlib_installed(), ...` lines  from `nilearn.plotting` and `nilearn.reporting` tests, as these tests are already ignored by `conftest.py`

I had not expected that but it makes sense to only have a single global config rather than one skipif by test.

File: nilearn/_utils/helpers.py (hunk 3 above)
Comment (Remi-Gau) [reply nilearn/_utils/helpers.py:9]: Once this is adressed I think we can merge the PR

**Review by Remi-Gau — APPROVED** [review]

**Pull request merged**